src/pybind/mgr/dashboard_v2/controllers/filesystem.py

```diff
@@ -0,0 +1,357 @@
+            pool_type = "metadata" if pool_id == metadata_pool_id else "data"
+            stats = pool_stats[pool_id]
+            pools_table.append({
+                "pool": pools[pool_id]['pool_name'],
+                "type": pool_type,
+                "used": stats['bytes_used'],
+                "avail": stats['max_avail']
+            })
+
+        standby_table = []
+        for standby in fsmap['standbys']:
+            metadata = self.mgr.get_metadata('mds', standby['name'])
+            mds_versions[metadata.get('ceph_version', 'unknown')].append(
+                standby['name'])
+
+            standby_table.append({
+                'name': standby['name']
+            })
+
+        return {
+            "filesystem": {
+                "id": fs_id,
+                "name": mdsmap['fs_name'],
+                "client_count": client_count,
+                "ranks": rank_table,
+                "pools": pools_table
+            },
+            "standbys": standby_table,
+            "versions": mds_versions
+        }
+
+    def _clients(self, fs_id):
+        cephfs_clients = self.cephfs_clients.get(fs_id, None)
+        if cephfs_clients is None:
+            cephfs_clients = CephFSClients(self.mgr, fs_id)
+            self.cephfs_clients[fs_id] = cephfs_clients
+
+        # try:
+        # status,
+        clients = cephfs_clients.get()
+        # except AttributeError:
+        #     raise cherrypy.HTTPError(404,
+        #                              "No filesystem with id {0}".format(fs_id))
+        if clients is None:
+            raise cherrypy.HTTPError(404,
+                                     "No filesystem with id {0}".format(fs_id))
+        # TODO do something sensible with status
+
+        # Decorate the metadata with some fields that will be
+        # indepdendent of whether it's a kernel or userspace
+        # client, so that the javascript doesn't have to grok that.
+        for client in clients:
+            if "ceph_version" in client['client_metadata']:
+                client['type'] = "userspace"
+                client['version'] = client['client_metadata']['ceph_version']
+                client['hostname'] = client['client_metadata']['hostname']
+            elif "kernel_version" in client['client_metadata']:
+                client['type'] = "kernel"
+                client['version'] = client['client_metadata']['kernel_version']
+                client['hostname'] = client['client_metadata']['hostname']
+            else:
+                client['type'] = "unknown"
+                client['version'] = ""
+                client['hostname'] = ""
+
+        return clients
+
+    def get_latest(self, daemon_type, daemon_name, stat):
+        data = self.mgr.get_counter(daemon_type, daemon_name, stat)[stat]
+        if data:
+            return data[-1][1]
+        else:
+            return 0
+
+    def format_dimless(self, n, width, colored=True):
```

Comment:
1. The API should not care about colours. I'd just remove it. 
2. The API should not care about formatting values in general. I'd just remove the whole method. That's a job for the frontend.

Reply:
done